spec/controllers/admin/posts_controller_spec.rb

```diff
@@ -0,0 +1,97 @@
+
+require 'rails_helper'
+
+describe Admin::PostsController do
+
+  before (:each) do
+    @user = FactoryGirl.create(:user)
+    @user.role = 'admin'
+    @user.save
+    sign_in @user
+  end
+
+  describe 'GET index' do
+    it 'assigns all admin_posts as @admin_posts' do
+      post = FactoryGirl.create(:post)
+      get :index
+      expect(assigns(:posts)).to eq([post])
+    end
+  end
+
+  describe 'GET show' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.